### PR TITLE
Fix brew pull with git commit.gpgsign enabled

### PR DESCRIPTION
**Pull request by alyssais**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Because of environment filtering, git tries to sign the commit but can't find gpg, which means that the patch won't be applied.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/pull.rb [hunk 1]
```diff
@@ -102,6 +102,18 @@ def pull
       ENV["GIT_COMMITTER_EMAIL"] = ENV["HOMEBREW_GIT_EMAIL"]
     end
 
+    # Depending on user configuration, git may try to invoke gpg.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/pull.rb]: Is there any way to identify this in advance rather than doing so unconditionally?

Comment (DomT4) [reply Library/Homebrew/dev-cmd/pull.rb]:
```ruby
Utils.popen_read("git config --get commit.gpgsign").chomp
```
Maybe?

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/pull.rb]: @alyssais any thoughts?

Comment (alyssais) [reply Library/Homebrew/dev-cmd/pull.rb]: What @DomT4 says looks good, though I'm not sure there's any harm in doing it unconditionally. I'll try to push that change tonight if we want it.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/pull.rb]: Yeh, that would be good, thanks 👍

**Comment by MikeMcQuaid** [comment]
@alyssais Thanks again!

**Pull request merged**